src/handlers/suggest/handler.js

```diff
@@ -1,84 +1,55 @@
-    const old_json = await new Promise((resolve, reject) => {
-        request({ uri: url }, function (error, response, body) {
-            if (error || response.statusCode !== 200) {
-                console.error(error);
-                console.error(`Responsecode: ${response.statusCode}`);
-                reject(`Request to ${url} failed`);
-            }
-            resolve(body);
-        });
-    });
-    return jsonDiff.diffString(JSON.parse(old_json), request_body.data, { color: '' });
-}
-
-async function messageBody(request_body) {
-    return `The following suggestion was submitted through the website:
-
-\`\`\`${request_body.new ? '' : 'diff'}
-${await getMessageContent(request_body)}
-\`\`\`
-
-**[Edit](https://company-json.netlify.com/#!doc=${encodeURIComponent(JSON.stringify(request_body.data))})**`;
-}
-
 async function suggest(request, h) {
     try {
         const request_body = request.payload;
 
-        const result = await octokit.issues.create({
-            owner: config.suggest.owner,
-            repo: config.suggest.repo,
-            title: request_body.new
-                ? 'New company suggestion' + (request_body.data.name ? ': `' + request_body.data.name + '`' : '')
-                : 'Suggested update for company `' + request_body.data.slug + '`',
-            // we create the issue just with the raw submission as body
-            // this is overwritten later on: the issue is added as source and it becomes a diff if this was an update
-            // but in case that second step fails, we have saved the raw data
-            body: JSON.stringify(request_body.data),
+        let files = {};
+        files['companies/' + request_body.data.slug + '.json'] = JSON.stringify(request_body.data, null, 4);
+
+        const title = request_body.new
+            ? 'New company suggestion' + (request_body.data.name ? ': `' + request_body.data.name + '`' : '')
+            : 'Suggested update for company `' + request_body.data.slug + '`';
+
+        const commit_msg = request_body.new
+            ? 'added ' + (request_body.data.name ? ': `' + request_body.data.name + '`' : '')
+            : 'updated `' + request_body.data.slug + '`';
+
+        const pr = await new Promise((resolve, reject) => {
+            octokit
+                .createPullRequest({
+                    owner: config.suggest.owner,
+                    repo: config.suggest.repo,
+                    base: config.suggest.branch,
+                    title: title,
+                    body: 'The following suggestion was submitted through the website:',
+                    head: 'suggest_' + request_body.data.slug + '_' + Date.now(),
+                    changes: [
+                        {
+                            files: files,
+                            commit: commit_msg,
+                        },
+                    ],
+                })
+                .then((pr) => resolve(pr))
+                .catch((e) => reject(e));
```

Comment:
I'm not quite sure how stable octokit-plugin-create-pull-request is (this quite a hard problem to solve!). Maybe we should have some sensible fallback (like creating an issue—doesn't have to be as fancy as previously, just dump the JSON) in case something goes wrong, so we don't lose the user's work in this case?